### PR TITLE
General api request errors

### LINKED ISSUE #80: Добавить проброс ошибок маппинга вместе с ответом сервера
В этот файл https://github.com/TouchInstinct/LeadKit/blob/master/Sources/Extensions/Alamofire/AlamofireRequest%2BExtensions.swift

Пример описания ошибки:
```swift
enum RequestError: Error {
    case connection(connectionError: Error)
    case mapping(mappingError: MapError, response: Any)
}
```

### PR DESCRIPTION
**Pull request by petropavel13**
closes #80 

```swift
extension RequestError: AlertRepresentable {

    public var title: String {
        return "Request error"
    }

    public var text: String {
        switch self {
        case .noConnection:
            return "No connection to the server."
        case .mapping, .invalidResponse:
            return "Invalid server response."
        case .network:
            return "Unknown network error."
        }
    }

    public var actionTitle: String {
        return "Retry"
    }

}
```

### CODE REVIEW

File: Sources/Protocols/ConfigurableView.swift [hunk 1]
```diff
@@ -23,17 +23,17 @@
 import Foundation
 
 /**
- *  protocol which ensures that specific type can create view model and can apply new view state with view model
+ *  Protocol that ensures that specific type can create view model and can apply new view state with view model
  */
-public protocol AbstractViewModelProtocol {
+public protocol ConfigurableView {
     associatedtype ViewModelType
```
Comment (igorkislyuk) [inline Sources/Protocols/ConfigurableView.swift]: Associated type should be renamed. Same about description for method.

File: Sources/Extensions/Alamofire/AlamofireRequest+Extensions.swift [hunk 2]
```diff
@@ -90,11 +92,60 @@ public extension Reactive where Base: DataRequest {
...
+}
+
+private extension ObservableType where E == ServerResponse {
+
+    func tryMapResult<R>(_ transform: @escaping (E) throws -> R) -> Observable<R> {
+        return map {
+            do {
+                return try transform($0)
+            } catch {
+                throw RequestError.mapping(error: error, response: $0.1)
```
Comment (igorkislyuk) [inline Sources/Extensions/Alamofire/AlamofireRequest+Extensions.swift]: Unnamed tuple :)

File: Sources/Enums/LeadKitError.swift [hunk 3]
```diff
@@ -22,7 +22,7 @@
 
 import Foundation
 
-/// Enum which represents common errors in LeadKit framework
+/// Enum that represents common errors in LeadKit framework
```
Comment (igorkislyuk) [inline Sources/Enums/LeadKitError.swift]: Is this error used anywhere?

Comment (petropavel13) [reply Sources/Enums/LeadKitError.swift]: `cast()` throws this error.

File: Sources/Extensions/Alamofire/AlamofireRequest+Extensions.swift (hunk 2 above)
Comment (petropavel13) [reply Sources/Extensions/Alamofire/AlamofireRequest+Extensions.swift]: It comes from RxAlamofire `responseResult(queue:responseSerializer:)`.
The only thing that I can do it:
```swift
.map { ServerResponse(response: $0.0, result: $0.1) }
```

File: Sources/Protocols/ConfigurableView.swift (hunk 1 above)
Comment (petropavel13) [reply Sources/Protocols/ConfigurableView.swift]: renamed to what? any suggestions?

Comment (igorkislyuk) [reply Sources/Protocols/ConfigurableView.swift]: Think about smth called `Configurator` instead of `ViewModelType` according to `ConfigurableView`.

**Pull request merged**